CMakeLists.txt

```diff
@@ -52,6 +52,17 @@ cmake_dependent_option(
 )
 option(BUILD_SHARED_LIBS "Build shared library" ON)
 
+option(ENABLE_CCACHE "Enable ccache for faster recompilation" ON)
+find_program(CCACHE_FOUND ccache)
+
+if(CCACHE_FOUND AND ENABLE_CCACHE)
+  set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCACHE_FOUND}")
+  set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK "${CCACHE_FOUND}")
```

Comment:
Don't use `RULE_LAUNCH_COMPILE` or `RULE_LAUNCH_LINK` for these. Since CMake 3.4, the right way to do this is to use the `CMAKE_<LANG>_COMPILER_LAUNCHER` variables instead. And `ccache` doesn't cache linker commands, so there's no need to use it as a launcher for the linker.

See updated explanation here: https://crascit.com/2016/04/09/using-ccache-with-cmake/#h-improved-functionality-from-cmake-3-4